internal/session/service_close_connections.go

```diff
@@ -0,0 +1,33 @@
+package session
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// CloseConnection is a domain service function that:
+// * closes requested connections
+// * uses the sessionId of the connection to see if the session meets conditions for termination
+func CloseConnections(ctx context.Context, sessionRepoFn *Repository, connectionRepoFn *ConnectionRepository,
+	closeWiths []CloseWith, opt ...Option) ([]closeConnectionResp, error) {
+	const op = "session.AuthorizeConnection"
+
+	closeInfos, err := connectionRepoFn.closeConnections(ctx, closeWiths)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+
+	for _, c := range closeInfos {
+		ses, _, err := sessionRepoFn.LookupSession(ctx, c.Connection.SessionId)
+		if err != nil {
+			return nil, err
+		}
+		if ses != nil {
+			version := ses.Version
+			sessionRepoFn.terminateSessionIfPossible(ctx, c.Connection.SessionId, version)
+		}
```

Comment:
I still don't follow why this version is needed, but I think if we need it, it should be done in `terminateSessionIfPossible` so it is part of the same transaction instead of calling LookupSession here.